src/utils/errors.rs

```diff
@@ -0,0 +1,114 @@
+use std::fmt;
+use thiserror::Error;
+use rocket::request::Request;
+use rocket::response::{self, Response, Responder};
+use rocket::http::{ContentType, Status};
+use serde_json;
+use serde::{Serialize, Deserialize};
+use std::io::Cursor;
+use log::debug;
+use anyhow::Result;
+
+pub type ApiResult<T, E = ApiError> = Result<T, E>;
+
+#[derive(Error, Debug, Serialize)]
```

Comment:
We should not need Serialize here

Reply:
hmmm somehow I do. Why do you expect it to not bee needed? I think it's necessary for the `Responder` implementation of `ApiError`

Reply:
Never mind that previous comment. You are right, once in the responder impl I use `self.details` instead, `Serialize` becomes totally unnecessary 👍